clients/vulnerable-code/src/main/kotlin/VulnerableCodeService.kt

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.clients.vulnerablecode
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.databind.json.JsonMapper
+import com.fasterxml.jackson.module.kotlin.registerKotlinModule
+
+import okhttp3.OkHttpClient
+
+import retrofit2.Retrofit
+import retrofit2.converter.jackson.JacksonConverterFactory
+import retrofit2.http.Body
+import retrofit2.http.GET
+import retrofit2.http.POST
+import retrofit2.http.Path
+
+/**
+ * Interface for a REST service that allows interaction with the VulnerableCode API to query information about
+ * vulnerabilities detected for specific packages.
+ */
+interface VulnerableCodeService {
+    companion object {
+        /**
+         * Create a new service instance that connects to the [serverUrl] specified and uses the optionally provided
+         * [client].
+         */
+        fun create(serverUrl: String, client: OkHttpClient? = null): VulnerableCodeService {
+            val vcClient = client ?: OkHttpClient()
+            val retrofit = Retrofit.Builder()
+                .client(vcClient)
```

Comment:
I prefer the `.apply { if (client != null) client(client) }` construct here like we do in other services to let RetroFit itself create the default client.

Reply:
Changed. However, as a future extension I was thinking about using a shared HTTP client for all our REST clients, which should be the optimized solution as described [here](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/). We could then have a client utils project that manages the shared HTTP client and provides some helper functionality for creating Retrofit services.

Reply:
Isn't that we effectively already do via the `OkHttpClientHelper`, which is the client that we use with all the services?

Reply:
I was not aware of this class. So, yes, this class could be used for this purpose.

However, if I read the referenced documentation for OkHttpClient correctly, it should not even be necessary to create multiple independent client instances, at least as long as a shared connection pool can be used. So the helper class could create a single client instance and use the `newBuilder()` function to create instances derived from this instance that can be further customized.

Reply:
Something like this? https://github.com/oss-review-toolkit/ort/pull/3562

Reply:
Nice. Is it okay to reference the `utils` package containing the `OkHttpClientHelper` class from projects under `clients`?

Reply:
No, unfortunately not, see the client's [README](https://github.com/oss-review-toolkit/ort/blob/master/clients/README.md). Clients should stay generic and not rely on any part of ORT, as they will eventually be moved to separate top-level projects for other to use, independently of ORT. But that's exactly why we allow to pass a client here, i.e. in our case one that was created via the `OkHttpClientHelper`.